### ISSUE TITLE
Visit installation with spack fails

### ISSUE THREAD

**Issue by thilinarmtb** [spack/spack#6406]
### Expected Result

I expected `spack install visit` to install visit on my system.

### Actual Result

I got the following error message:
```
Error: Conflicts in concretized spec "visit@2.12.2%gcc@6.3.0 build_type=RelWithDebInfo arch=linux-ubuntu17.04-x86_64 /yvx6trz"
```

### Steps to reproduce the issue

```console
$ spack install visit
```

### Information on your system

I am using Ubuntu 17.04 and the default `packages.yaml` and `modules.yaml` files
(I did not change these after cloning visit from github). I am using the latest `develop`
branch as of now (commit id: d4ab511f7).

**Comment by adamjstewart** [comment]
Short answer: This is a concretization bug. To get around it for now, you can run:
```
$ spack install visit ^netcdf~mpi ^hdf5~mpi
```
Long answer: Spack has trouble backtracking far enough up the DAG to override default variants: http://spack.readthedocs.io/en/latest/known_issues.html#variants-are-not-properly-forwarded-to-dependencies

The problem in this case is that `visit` explicitly requires `hdf5~mpi`, but it also depends on `vtk`, which depends on `netcdf`. The default for `netcdf` is `+mpi`, which requires `hdf5+mpi`. These things conflict, creating the error message you see. You can get a more verbose error message by running:
```
$ spack spec visit
...
"+mpi" conflicts with "netcdf^hdf5~mpi" [netcdf+mpi requires hdf5+mpi]
```
Aside: @alalazo Can we get the same conflict error message for `spack install`?

It looks like @eschnett added the `hdf5~mpi` dependency in #1762. @eschnett Is there any way we can get this to build with `hdf5+mpi`? Removing the `~mpi` constraint will prevent the concretization problem from arising.

**Comment by thilinarmtb** [comment]
Thank you very much @adamjstewart for the reply. I will try to install and test visit this way.

**Comment by adamjstewart** [comment]
Fixed by #6947.
